projects/wallet-service-reliable-integration/0001-design.md

```diff
@@ -0,0 +1,205 @@
+# Summary
+
+The goal of this document is to describe a new method of syncing with a Hathor node, using the reliable integrations feature and replace the current sync daemon on our production services
+
+# Motivation
+
+At present, our wallet-service sync daemon operates by utilizing the fullnode's WebSocket and HTTP APIs to maintain synchronization with the fullnode. With this approach, we have to manually handle reorganizations, making it not very maintenance friendly as we have to deal with all potential edge cases that are already handled in the fullnode, again in the daemon.
+  
+It was precisely with such applications in mind that we developed our '[Reliable Integrations](https://github.com/HathorNetwork/rfcs/blob/master/projects/reliable-integration/0001-high-level-design.md)' feature. The concept behind this feature is to implement a reliable, integrated event management system. Through this system, a series of events would be dispatched in the exact order of their occurrence, ensuring that if we follow this exact series of events, the same state will be obtained
+
+# Reference-level explanation
```

Comment:
If I got it right, you are always running with a window of size 1. Is that right? If yes, please, make sure to add this piece of information to the design.

Reply:
Great! Thanks!